compiler/hlint-testing/src/Main.hs

```diff
@@ -0,0 +1,167 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- A little program to help prove the use of hlint with DAML.
+
+{-# OPTIONS_GHC -Wno-missing-fields #-}
+
+module Main (main) where
+
+import "ghc-lib-parser" HsSyn
+import "ghc-lib-parser" Config
+import "ghc-lib-parser" DynFlags
+import "ghc-lib-parser" Platform
+import "ghc-lib-parser" StringBuffer
+import "ghc-lib-parser" Fingerprint
+import "ghc-lib-parser" Lexer
+import "ghc-lib-parser" ErrUtils
+import qualified "ghc-lib-parser" Parser
+import "ghc-lib-parser" FastString
+import "ghc-lib-parser" Outputable
+import "ghc-lib-parser" SrcLoc
+import "ghc-lib-parser" Panic
+import "ghc-lib-parser" HscTypes
+import "ghc-lib-parser" HeaderInfo
+import "ghc-lib-parser" ApiAnnotation
+import "ghc-lib-parser" GHC.LanguageExtensions.Type
+
+import Control.Monad
+import Control.Monad.Extra
+import System.FilePath
+import System.Environment
+import System.IO.Extra
+import qualified Data.Map as Map
+import Data.List
+
+import Language.Haskell.HLint4
+
+{- To test, create a file e.g. ~/Test.daml with contents
+   ```
+   daml 1.2
+   module Main where
+
+   main : IO ()
+   main = print $ 3 + 4
+   ```
+   then, bazel run //compiler/hlint-testing:hlint-test ~/Test.daml.
+
+   For the above module, we expect output like,
+   ```
+   [/Users/shaynefletcher/Test.daml:1:1: Ignore: Use module export list
+    Found:
+      module Main where
+    Perhaps:
+      module Main (
+              module Main
+          ) where
+    Note: an explicit list is usally better
```

Comment:
```suggestion
    Note: an explicit list is usually better
```

Reply:
Ha. You found a `hlint` bug (probably of my making though) :)